sql/src/main/java/io/crate/metadata/Reference.java

```diff
@@ -163,6 +198,24 @@ public Integer position() {
         return position;
     }
 
+    @Nullable
+    public String formattedDefaultExpression() {
+        return formattedDefaultExpression;
+    }
+
+    public boolean hasDefaultExpression() {
```

Comment:
Static analysis tools usually can't infer the relationship between `hasDefaultExpression = true == formattedDefaultExpression != null` so I'd tend to remove this method and work with the null checks directly.